api/internal/localizer/locloader.go

```diff
@@ -117,7 +117,8 @@ func (ll *locLoader) Load(path string) ([]byte, error) {
 // Otherwise, New returns an error.
 func (ll *locLoader) New(path string) (ifc.Loader, error) {
 	repoSpec, err := git.NewRepoSpecFromURL(path)
```

Comment:
Lines 119-123 are a defensive check. 

The `ll.Loader.New(path)` on line 125 will try to `git fetch` the url. If there is no `ref`, this could take quite some time, potentially timing out the integration tests. As a result, I moved the check for `ref` before the call to `New()`.

Actually, though, there is a problem with the current implementation when `path` is a file url. I was going to introduce the change with a file url integration test in a separate PR. I can introduce that change here if you think it makes more sense. Please reference: https://github.com/kubernetes-sigs/kustomize/pull/4797/files#diff-3da63a4a59b5a528cb33383c63a905142e509f2c229d2fb4703c22bd008fb9f2R113-R120